rgw/v2/tests/s3_swift/test_object_lock.py

```diff
@@ -77,13 +76,27 @@ def test_exec(config, ssh_con):
                     Bucket=bucket_name_to_create, ObjectLockEnabledForBucket=True
                 )
                 # put object lock configuration for bucket
-                s3_conn_client.put_object_lock_configuration(
-                    Bucket=bucket_name_to_create,
-                    ObjectLockConfiguration={
-                        "ObjectLockEnabled": "Enabled",
-                        "Rule": {"DefaultRetention": {"Mode": "COMPLIANCE", "Days": 1}},
-                    },
-                )
+                if config.test_ops["compliance_mode"] is True:
+                    s3_conn_client.put_object_lock_configuration(
+                        Bucket=bucket_name_to_create,
+                        ObjectLockConfiguration={
+                            "ObjectLockEnabled": "Enabled",
+                            "Rule": {
+                                "DefaultRetention": {"Mode": "COMPLIANCE", "Days": 1}
+                            },
+                        },
+                    )
+                else:
+                    s3_conn_client.put_object_lock_configuration(
+                        Bucket=bucket_name_to_create,
+                        ObjectLockConfiguration={
+                            "ObjectLockEnabled": "Enabled",
+                            "Rule": {
+                                "DefaultRetention": {"Mode": "GOVERNANCE", "Days": 1}
+                            },
+                        },
+                    )
+
```

Comment:
@TejasC88 , Can we define a function for the put_object_lock call in the resuable and call it here?

Reply:
@viduship updated ,pls review